### PR TITLE
add cadvisor as exporter

### PR DESCRIPTION
**Pull request by YanjieGao**
deploy cadvisor to collect container level metrics.
This deployment support get container label and container gpu metrics.

### CODE REVIEW

**Comment by ydye** [comment]
After this pr, should we still need aggregating kube c-advisor from prom?

**Comment by ydye** [comment]
And we should make the user know that the ca port shouldn't be the default port because of the ca in kubelet.

**Comment by YanjieGao** [comment]
Kubelet use 4194 as its default ca port. ca docker use 8089 as default port. We could add a comment at the above the config

**Comment by YanjieGao** [comment]
After this PR, prometheus will aggregate ca metrics from PAI deployed ca. Do not use kubelet's ca.

**Comment by ydye** [comment]
So can we close another PR?

**Comment by YanjieGao** [comment]
This PR is export ca metrics. Another PR is prom fetch ca metrics. Not duplicate.

File: service-deployment/bootstrap/prometheus/node-exporter-ds.yaml.template [hunk 1]
```diff
@@ -55,6 +67,36 @@ spec:
         - mountPath: {{ clusterinfo[ 'dataPath' ] }}/prometheus
           name: collector-mount
         name: gpu-exporter
+      - image: google/cadvisor:latest
```
Comment (ydye) [inline service-deployment/bootstrap/prometheus/node-exporter-ds.yaml.template]: Do not use latest.

Comment (YanjieGao) [reply service-deployment/bootstrap/prometheus/node-exporter-ds.yaml.template]: ok, fix

File: service-deployment/bootstrap/prometheus/node-exporter-ds.yaml.template [hunk 2]
```diff
@@ -55,6 +67,36 @@ spec:
         - mountPath: {{ clusterinfo[ 'dataPath' ] }}/prometheus
           name: collector-mount
         name: gpu-exporter
+      - image: google/cadvisor:v0.29.0
```
[the remark this answers is not in the document]
Comment (YanjieGao) [reply service-deployment/bootstrap/prometheus/node-exporter-ds.yaml.template]: yes, hostnet config is at the bottom of the file

**Comment by ydye** [comment]
Merge it, after passing test.

File: service-deployment/clusterconfig-example.yaml [hunk 3]
```diff
@@ -117,7 +115,9 @@ clusterinfo:
     prometheus_port: 9090
     # port for node exporter
     node_exporter_port: 9100
-  
+    # cadvisor port
```
Comment (ydye) [inline service-deployment/clusterconfig-example.yaml]: Why put this configuration in grafana?

Comment (YanjieGao) [reply service-deployment/clusterconfig-example.yaml]: Put it in promethues. This config currently is used by prometheus. The web shows the config is folded.

Comment (ydye) [reply service-deployment/clusterconfig-example.yaml]: Sorry, I didn't expand all the code. 😆

**Comment by YanjieGao** [comment]
pass test

**Comment by YanjieGao** [comment]
pass test

**Pull request merged**